File: test/new_lang/test7/TestProgram.pas
program TestProgram;
uses SplashKit;

procedure Main();
var
  v: Flag;
  v1: BasicFlag;
begin
  v1 := GetBasicFlag();
  PrintBasicFlag(v1);

  PrintBasicFlag(OPTION_1);
  PrintBasicFlag(OPTION_2);
  PrintBasicFlag(A_OPTION);

  WriteLn('---');
  v := GetFlag();
  PrintFlag(v);

  PrintFlag(FLAG_1);
  PrintFlag(FLAG_10);
end;

begin
  Main();
end.

File: test/new_lang/test7/TestProgram.cs
using static SplashKitSDK.SplashKit;
using static System.Console;
using System.Collections.Generic;

public class Program
{
  public static void Main()
  {
    Flag v;
    BasicFlag v1;

    v1 = GetBasicFlag();
    PrintBasicFlag(v1);

    PrintBasicFlag(BasicFlag.Option1);
    PrintBasicFlag(BasicFlag.Option2);
    PrintBasicFlag(BasicFlag.AOption);

    WriteLine("---");
    v = GetFlag();
    PrintFlag(v);

    PrintFlag(Flag.Flag1);
    PrintFlag(Flag.Flag10);
  }
}
